documentation/general/eliminate-duplicate-file.md

```diff
@@ -0,0 +1,302 @@
+This is critical for resolving version conflicts.
+Higher version references may cause load failures, while lower version references could trigger security alerts.
+Compile-time checks can be added to enforce the desired version rules. The VMR has helped reduce version conflict occurrences.
+
+The upcoming [NuGet Vision 2027 work](https://microsoft-my.sharepoint.com/:w:/r/personal/aortiz_microsoft_com/Documents/NuGet%20Vision%202027.docx?d=w9f413c3dc36a4e7d887fe007071e10c0&csf=1&web=1&e=MD76Wr&nav=eyJoIjoiMjE2NDM5NDIwIn0) for "better supporting apps running in hosted environments or with specific composition/deployment patterns" will help facilitate version conflict resolution in this area. This work will introduce the capability to declare dependencies as "provided by the hosting environment," allowing components to use the host environment's version rather than carrying their own copy. This approach will make it much easier to eliminate version differences and consolidate on shared dependency versions within the SDK.
+
+Similar patterns will be used for architecture differences as discussed in the [Architecture Differences](#architecture-differences) section.
+
+### Testing Strategy
+
+Once duplicated assemblies are removed, a regression test should:
+
+- Detect duplicate files in the SDK layout.
+- Fail the build if duplicates are found.
+- Prevent regressions.
+
+## Proof of Concept Results
+
+A proof of concept was implemented for `dotnet-watch` and `dotnet-format`, two of the largest sources of duplication.
+The POC used the **AssemblyLoadContext** approach with a shared assembly location and yielded the following results:
+
+| Metric       | Baseline 11.0 (SDK only) | Size Reduction | % Reduction |
+|--------------|--------------------------|----------------|-------------|
+| Archive Size | 100 MB                   | 21 MB          | 21.0%       |
+| Disk Size    | 296 MB                   | 62 MB          | 20.8%       |
+| File Count   | 3,957                    | 384 files      | 9.7%        |
+
+**Note:** These numbers are from a Linux development build, which differs significantly from official signed/optimized builds included in the [By the Numbers](#by-the-numbers) section.
+
+## Other Concerns
+
+### Architecture Differences
+
+As noted in the [duplicate categorization](#linux-x64), a portion of duplicates with the same version but different hash are due to architecture differences (AnyCPU vs x64).
+The plan is to eliminate these differences by standardizing on CPU-specific versions.
+
+Initial analysis indicates these differences stem from AnyCPU builds coexisting with CPU-specific builds of the same assembly.
+The general approach will be to prefer the CPU-specific version over the AnyCPU version when eliminating these duplicates, as CPU-specific builds can offer better performance characteristics for the target platform.
```

Comment:
They do not. .NET Core only supports IL-only binaries. 

If you are referring to R2R binaries, that's a completely different matter.